docs/deploy_hostack_on_psi.md

```diff
@@ -0,0 +1,274 @@
+    **hotloop**: Provides a generic looping mechanism to execute sequences of commands, apply Kubernetes manifests (Custom Resources - CRs), and implement wait conditions, facilitating complex automation workflows.
+    **redfish_virtual_bmc**: Deploys the sushy-emulator, a RedFish Virtual BMC (Baseboard Management Controller) service, typically onto the OpenShift cluster. This is often crucial for managing virtualized "bare metal" nodes as required by RHOSO components like OpenStack Ironic.
+
+
+* **Key Ansible Playbooks**:
+    The deployment process is orchestrated by a sequence of Ansible playbooks, with bootstrap.yml serving as the main entry point. This master playbook imports other playbooks to execute specific stages of the deployment:
+    1. **01-infra.yml**: Provisions the virtual infrastructure on the OpenStack cloud.
+    2. **02**-**bootstrap_controller**.yml: Prepares and bootstraps the controller node.
+    3. **03**-**install_ocp**.yml: Installs the OpenShift Container Platform cluster.
+    4. **04_redfish_virtual_bmc**.yml: Deploys the sushy-emulator (RedFish Virtual BMC).
+    5. **05_deploy_rhoso**.yml: Deploys the Red Hat OpenStack overcloud (RHOSO).
+
+---
+# 2. Preparing Your OpenStack Environment and Client Machine
+
+- Successful HotStack deployment depends on OpenStack cloud environment and the client machine from which deployment will be orchestrated.
+- It is recommended to spin up the small client instance which will be used for deployment.
+- So we have to meticulous steps to meet prerequisites or else it will lead to deployment failure.
+
+## 2.1 Client-Machine
+- It is recommended to create a small instance on OpenStack Cloud(PSI) to run the deployment script.
+- Recommended flavor and OS:
+    - g.standard.small
+    - CentOS-Stream-9-latest
+## 2.2 Client-Side Tooling (Machine Running Ansible)
+After SSH into Client Machine we need tools to run the OpenStack
+
+* **Ansible:** A HotStack is built upon Ansible, a working Ansible installation is mandatory.
+    * We can install ansible using dnf package manager
+
+```
+sudo dnf install -y ansible-core
+```
+or with pip
+
+```bash
+mkdir -p ~/ansible-venv
+python3 -m venv ~/ansible-venv
+source ~/ansible-venv/bin/activate
+pip install ansible
+```
+
+* **Required Ansible Collections**:
+    - HotStack depends on specific Ansible collections that provide modules for interacting with OpenStack and performing cryptographic operations:
+```bash
+ansible-galaxy collection install community.crypto
+ansible-galaxy collection install openstack.cloud
+```
+
+- **OpenStack Client**: The openstack command-line interface (CLI) client is crucial for various preparatory tasks
+    - To install OpenStack Client we need to configure:
+
+```bash
+sudo dnf config-manager --enable crb
+sudo dnf install -y centos-release-openstack-dalmatian.noarch
+```
+
+```bash
+sudo dnf install -y python-openstackclient
+```
+
+**Heat Client**:
+
+```bash
```

Comment:
Should we compress the step for openstack client and heat client into one step, and have the example command like this:

```
sudo dnf install -y python-openstackclient python-heatclient

```